src/Module/Magic.php

```diff
@@ -65,120 +65,102 @@ public function __construct(App $app, L10n $l10n, App\BaseURL $baseUrl, App\Argu
 
 	protected function rawContent(array $request = [])
 	{
-		$this->logger->info('magic module: invoked');
+		if ($_SERVER['REQUEST_METHOD'] == 'HEAD') {
+			$this->logger->debug('Got a HEAD request');
+			System::exit();
+		}
 
-		$this->logger->debug('args', ['request' => $_REQUEST]);
+		$this->logger->debug('Invoked', ['request' => $request]);
 
 		$addr  = $request['addr'] ?? '';
 		$dest  = $request['dest'] ?? '';
 		$bdest = $request['bdest'] ?? '';
 		$owa   = intval($request['owa'] ?? 0);
-		$cid  = 0;
 
-                // bdest is preferred as it is hex-encoded and can survive url rewrite and argument parsing
+		// bdest is preferred as it is hex-encoded and can survive url rewrite and argument parsing
 		if (!empty($bdest)) {
 			$dest = hex2bin($bdest);
-			$this->logger->info('bdest detected. ', ['dest' => $dest]);
+			$this->logger->debug('bdest detected', ['dest' => $dest]);
 		}
-		if (!empty($addr)) {
-			$cid = Contact::getIdForURL($addr);
-		} elseif (!empty($dest)) {
-			$cid = Contact::getIdForURL($dest);
+
+		if (!empty($addr ?: $dest)) {
+			$contact = Contact::getByURL($addr ?: $dest);
 		}
```

Comment:
```suggestion
		$contact = Contact::getByURL($addr ?: $dest);
```

Reply:
This would lead to unneeded logging in `getIdForURL`. We have a check there for an empty url, but since this is unexpected there, we log it with `Empty url, quitting`. This check prevents this.

Reply:
```suggestion
		if ($addr ?: $dest) {
			$contact = Contact::getByURL($addr ?: $dest);
		}
```